builder/src/js/isValidate.js

```diff
@@ -1,32 +1,31 @@
-import util from './util.js';
+import { util, brackets } from './util.js';
 
-const [SBRACKETS_LEFT, SBRACKETS_RIGHT] = ["(",")"];
-const [MBRACKETS_LEFT, MBRACKETS_RIGHT] = ["{","}"];
-const [BBRACKETS_LEFT, BBRACKETS_RIGHT] = ["[","]"];
+const { SBRACKETS_LEFT, SBRACKETS_RIGHT, MBRACKETS_LEFT, MBRACKETS_RIGHT, BBRACKETS_LEFT, BBRACKETS_RIGHT } = brackets;
 let BRACKETS_SET = [SBRACKETS_LEFT,SBRACKETS_RIGHT, MBRACKETS_LEFT, MBRACKETS_RIGHT, BBRACKETS_LEFT, BBRACKETS_RIGHT];
 let BRACKETS_RIGHT = [SBRACKETS_RIGHT, MBRACKETS_RIGHT, BBRACKETS_RIGHT]
+let isValid = true;
 
 const isValidate = (str) => {
-    let isValid = true;
+    isValid = true;
     const bracketStack = [];
-    const letterArr = util.strToLetter(str);
-    const bracketArr = letterArr.filter(el => BRACKETS_SET.includes(el));
+    const bracketArr = util.strToLetter(str).filter(el => BRACKETS_SET.includes(el));
     bracketArr.forEach(ele => {
         if(BRACKETS_RIGHT.includes(ele)){
-            const lastEl = bracketStack[bracketStack.length - 1];
-            switch(lastEl) {
-                case SBRACKETS_LEFT:
-                    ele !== SBRACKETS_RIGHT ? isValid =  false : "";
-                case MBRACKETS_LEFT:
-                    ele !== MBRACKETS_RIGHT ? isValid =  false : "";
-                case BBRACKETS_LEFT:
-                    ele !== BBRACKETS_RIGHT ? isValid =  false : "";
-            }
+            bracketPairCheck(ele, bracketStack[bracketStack.length - 1])
+            bracketStack.pop();
         } else {
             bracketStack.push(ele);
         }
     });
+    if(bracketStack.length !== 0) isValid = false;
     return isValid;
 }
 
+function bracketPairCheck(RightBracket, lastEl){
+    RightBracket === SBRACKETS_RIGHT && lastEl !== SBRACKETS_LEFT ? isValid = false : "";
+    RightBracket === MBRACKETS_RIGHT && lastEl !== MBRACKETS_LEFT ? isValid = false : "";
+    RightBracket === BBRACKETS_RIGHT && lastEl !== BBRACKETS_LEFT ? isValid = false : "";
+}
```

Comment:
P4
isValid 변수는 isValidate 함수 내부에서만 사용되도록 ture/false 를 반환하는 함수로 변경하면 좋을 것 같아요.
```suggestion
function bracketPairCheck(bracket, lastEl){
    if (bracket === SBRACKETS_RIGHT) {
        return lastEl === SBRACKETS_LEFT;
    }
    if (bracket === MBRACKETS_RIGHT) {
        return lastEl === MBRACKETS_LEFT;
    }
    if (bracket === BBRACKETS_RIGHT) {
        return lastEl === BBRACKETS_LEFT;
    }
    reutrn false;
}
```